### PR TITLE
Edit Services doc content

### PR DESCRIPTION
**Pull request by willingc**
Primarily edits for user clarity and additional links to examples.

Added links to major sections to ease user navigation.

Reflowed some text to make easier for doc edits.

### CODE REVIEW

File: docs/source/services.md [hunk 1]
```diff
@@ -283,7 +320,7 @@ and taking note of the following process:
     where cookie-value is the url-encoded value of the `jupyterhub-services` cookie.
     This request must be authenticated with a Hub API token in the `Authorization` header.
     For example, with [requests][]:
```
Comment (parente) [inline docs/source/services.md]: Missing link?

File: docs/source/services.md [hunk 2]
```diff
@@ -301,10 +338,10 @@ and taking note of the following process:
...
 
    ```json
-   { 
+   {
      "name": "inara",
      "groups": ["serenity", "guild"],
-     
+
    }
    ```
```
Comment (parente) [inline docs/source/services.md]: Worth adding a section at the bottom about the nbviewer implementation? It's not very instructive, but it is a an example of something you might conceivably want to secure behind Hub auth.

* https://github.com/jupyter/nbviewer#securing-the-notebook-viewer
* https://github.com/jupyter/nbviewer/blob/master/nbviewer/providers/base.py#L94

File: docs/source/services.md (hunk 1 above)
Comment (parente) [inline docs/source/services.md]: Ah. My bad. Missed it there.

File: docs/source/services.md (hunk 2 above)
Comment (parente) [inline docs/source/services.md]: Thank you for adding it! 🍰  + 🍰 

One clarification: nbviewer can be run either as a hub managed or externally managed service. The README for nbviewer does say it. Not sure if it's worth clarifying in the Hub docs here.

Comment (willingc) [inline docs/source/services.md]: Will add both. Thanks!

**Pull request merged**